### PR TITLE
feat(wip): forms

### PR DESCRIPTION
**Pull request by heydoyouknowme0**

### CODE REVIEW

**Review by GetPsyched — CHANGES_REQUESTED** [review]

File: prisma/migrations/20240218193937_form_related_changes/migration.sql
[the remark this answers is not in the document]
Comment (heydoyouknowme0) [reply prisma/migrations/20240218193937_form_related_changes/migration.sql:1]: Then forms just won't work.

File: actions/form.actions.tsx [hunk 1]
```diff
@@ -1,18 +1,23 @@
 'use server';
-import FormEditNotAllowed from '~/components/forms/FormEditNotAllowed';
-import FormExpired from '~/components/forms/FormExpired';
-import FormNotFound from '~/components/forms/FormNotFound';
-import FormSingleResponse from '~/components/forms/FormSingleResponse';
+import { revalidatePath } from 'next/cache';
```
Comment (GetPsyched) [inline actions/form.actions.tsx]: There should be a line break after `use server`:
```ts
'use server';

import { revalidatePath } from 'next/cache';
```

**Review by GetPsyched — CHANGES_REQUESTED** [review]

File: prisma/migrations/20240218193937_form_related_changes/migration.sql
[the remark this answers is not in the document]
Comment (GetPsyched) [reply prisma/migrations/20240218193937_form_related_changes/migration.sql:1]: No I mean, we will migrate, we just won't push the migration scripts to the Git repo.

File: .gitignore [hunk 2]
```diff
@@ -38,3 +38,5 @@ yarn-error.log*
 
 # typescript
 *.tsbuildinfo
+.vscode
+.vscode
```
Comment (GetPsyched) [inline .gitignore:42]: Don't commit this.

File: .eslintrc.json [hunk 3]
```diff
@@ -39,6 +39,7 @@
         "ignoreMemberSort": false
       }
     ],
+
```
Comment (GetPsyched) [inline .eslintrc.json:42]: Shouldn't be included in this PR

File: app/[locale]/forms/layout.tsx [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+function layout({ children }: { children: ReactNode }) {
+  return (
+    <>
+      {children}
+      <Toaster />
+    </>
+  );
+}
+
+export default layout;
```
Comment (GetPsyched) [inline app/[locale]/forms/layout.tsx:14]: No need to export in a different line, just do `export default function Layout`. Also, you should put a WIP `page.tsx` here instead of making a layout, since there will be a default forms landing page in the future.

File: components/forms/FormEditNotAllowed.tsx
Comment (GetPsyched) [inline components/forms/FormEditNotAllowed.tsx:1]: All of these components are just different error messages with the exact same structure:
```html
<div>
  <h1></h1>
  <p></p>
</div>
```
Instead, just make a generic component that accepts a `title` and `subtitle`/`content` to which you can pass the strings to.

File: components/forms/FormSubmitFormFinal.tsx [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+  };
+  const onSubmit = async () => {
+    console.log(forms.getValues());
+    const output = await forms.trigger();
+    console.log(forms.getValues());
+    console.log(output);
+    if (!output) return;
+    const result = await submitForm(form.id, forms.getValues());
+    toast(result);
+    Router.push('/en/forms');
```
Comment (GetPsyched) [inline components/forms/FormSubmitFormFinal.tsx:111]: Typo, I think you missed making this dynamic.
```suggestion
    Router.push(`/${locale}/forms`);
```

File: components/forms/FormSubmitFormFinal.tsx [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+    if (currentStep > 0) {
+      setPreviousStep(currentStep);
+      setCurrentStep((step) => step - 1);
+    }
+  };
+  const onSubmit = async () => {
+    console.log(forms.getValues());
+    const output = await forms.trigger();
+    console.log(forms.getValues());
+    console.log(output);
```
Comment (GetPsyched) [inline components/forms/FormSubmitFormFinal.tsx:107]: A bunch of console logs in this file, probably others too. Clear them off if testing is done.

**Review by GetPsyched — CHANGES_REQUESTED** [review]

File: components/forms/FormEditNotAllowed.tsx
Comment (heydoyouknowme0) [reply components/forms/FormEditNotAllowed.tsx:1]: Actually they were made as placeholders, but if this much's fine then ok

File: .gitignore (hunk 2 above)
Comment (heydoyouknowme0) [reply .gitignore:42]: this branch is based upon genericInputs, I have informed aditya will cherry pick changes once committed

File: components/forms/FormEditNotAllowed.tsx
Comment (GetPsyched) [reply components/forms/FormEditNotAllowed.tsx:1]: Placeholders for what?

**Comment by GetPsyched** [comment]
Closed in favour of #112
